agent/config/builder.go

```diff
@@ -1562,6 +1562,7 @@ func (b *builder) checkVal(v *CheckDefinition) *structs.CheckDefinition {
 		Body:                           stringVal(v.Body),
 		DisableRedirects:               boolVal(v.DisableRedirects),
 		TCP:                            stringVal(v.TCP),
+		UDP:                            stringVal(v.UDP),
```

Comment:
I don't see `agent/config/builder.go` mentioned in the developer docs that type to capture all the places that need to be checked for health check changes. Can you mention the config builder file? Maybe it can just be included in lines 4 and 5 (which both reference `agent/config/config.go`, which I think rely on this builder.go file).

Reply:
I realize I didn't link to the dev docs I was referencing 🤦 

https://github.com/hashicorp/consul/blob/main/docs/service-discovery/health-checks.md#check-registration-flows

Reply:
@jkirschner-hashicorp , thanks. The doc is updated; please take a look.